pkg/fixtures/vmindexreport/generator.go

```diff
@@ -0,0 +1,241 @@
+// Package vmindexreport provides utilities for generating fake VM index reports
+// for testing and load testing purposes.
+package vmindexreport
+
+import (
+	"fmt"
+	"math/rand"
+	"regexp"
+	"strconv"
+	"strings"
+
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	v1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+)
+
+const (
+	// MockDigest is kept in sync with pkg/virtualmachine/enricher/enricher_impl.go
+	MockDigest = "sha256:900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900d"
+
+	// MockDigestWithRegistry is the digest with a mock registry prefix for use with name.NewDigest()
+	MockDigestWithRegistry = "vm-registry/repository@sha256:900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900d"
+)
+
+// numericRegex matches sequences of digits in a version string.
+var numericRegex = regexp.MustCompile(`\d+`)
+
+// NormalizeRPMVersion parses an RPM version string and returns a 10-element int32 slice.
+// Only the first 3 numeric components (major, minor, patch) are extracted since those
+// are what matter for vulnerability matching. The rest are zeros.
+// The epoch prefix (e.g., "1:") is stripped before parsing.
+// Example: "1:1.54.0-3.el9_7" → [1, 54, 0, 0, 0, 0, 0, 0, 0, 0]
+func NormalizeRPMVersion(version string) []int32 {
+	result := make([]int32, 10)
+
+	// Strip epoch prefix if present (e.g., "1:" in "1:1.54.0-3.el9_7")
+	if idx := strings.Index(version, ":"); idx != -1 {
+		version = version[idx+1:]
+	}
+
+	// Extract first 3 numeric sequences from the version string
+	matches := numericRegex.FindAllString(version, 3)
+
+	for i, match := range matches {
+		if val, err := strconv.ParseInt(match, 10, 32); err == nil {
+			result[i] = int32(val)
+		}
+	}
+
+	return result
+}
+
+// Generator generates fake VM index reports using a constant pre-built template.
+type Generator struct {
+	packages     map[string]*v4.Package
+	repositories map[string]*v4.Repository
+	environments map[string]*v4.Environment_List
+}
+
+// selectPackageIndices returns a slice of indices into Rhel9Packages based on numRequested.
+// - numRequested <= 0: returns all indices sequentially [0, 1, ..., totalAvailable-1]
+// - numRequested < totalAvailable: randomly samples numRequested indices
+// - numRequested >= totalAvailable: uses all indices, then duplicates randomly to fill
+func selectPackageIndices(rng *rand.Rand, numRequested, totalAvailable int) []int {
```

Comment:
I am considering whether changing the randomness based on how `numRequested` compares to `totalAvailable` is necessary. Now, if it is exactly `totalAvailable` we will always get the same values, if it's larger we will get some random values too, and if it's smaller we'll get completely random values.

How about we make this simpler and always return random data? e.g.:

```go
if numRequested <= 0 {
  numRequested = totalAvailable
}

return rng.Perm(totalAvailable)[:numRequested]
```

Reply:
I would prefer that the function behaves in the same way (always random/always verbatim) regardless of `numRequested` v `totalAvailable`, but that's just for coherence. I understand it does not matter in this case and I agree we have more important things to do, so I am happy with this version if you prefer to keep it like this

Reply:
I understand. Based on what you wrote about the preferences, I would propose to address that in one of the follow-ups. Would you be okay opening a PR for this?